src/luarocks/fs/lua.lua

```diff
@@ -157,6 +157,53 @@ function fs_lua.dir(at)
    return coroutine.wrap(function() fs.dir_iterator(at) end)
 end
 
+--- List the Lua modules at a specific require path.
+-- eg. `modules("luarocks.cmd")` would return a list of all LuaRocks command
+-- modules, in the current Lua path.
+function fs_lua.modules(at)
+   at = at or ""
+   if #at > 0 then
+      -- turn require path into file path
+      at = at:gsub("%.", package.config:sub(1,1)) .. package.config:sub(1,1)
+   end
+
+   local path = package.path:sub(-1, -1) == ";" and package.path or package.path .. ";"
+   local paths = {}
+   for location in path:gmatch("(.-);") do
+      location = location:match("^%s*(.-)%s*$")  -- trim whitespace
```

Comment:
does package.path items ever have whitespace? I don't think it's allowed...

Reply:
I had LuaJIT available and it does...
```
$ LUA_PATH="/some/dir with;spaces/ " luajit
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/
JIT: ON SSE2 SSE3 SSE4.1 BMI2 fold cse dce fwd dse narrow loop abc sink fuse
> print(package.path)
/some/dir with;spaces/
> print(package.path .."-")
/some/dir with;spaces/ -
>
```

So I guess it can be done. Not wise, but can be done.

Reply:
But then LUA_PATH is misconfigured: Lua/LuaJIT will use the space like any other character (after all, filenames can _end_ with spaces, as crazy as that may sound):

```
LUA_PATH="$LUA_PATH;/foo bar/?.lua;/bla luajit
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/
JIT: ON SSE2 SSE3 SSE4.1 BMI2 fold cse dce fwd dse narrow loop abc sink fuse
> require"foo"
stdin:1: module 'foo' not found:
	no field package.preload['foo']
	no file './foo.lua'
	no file './foo/init.lua'
	no file './foo.lua'
	no file '/usr/local/share/lua/5.1/foo.lua'
	no file '/usr/local/share/lua/5.1/foo/init.lua'
	no file '/foo bar/foo.lua'
	no file '/bla '
	no file '/usr/local/lib/lua/5.1/foo.so'
	no file './foo.so'
stack traceback:
	[C]: in function 'require'
	stdin:1: in main chunk
	[C]: at 0x00404c30
```

Spot the space in `'bla '`, it is simply assumed to be part of the name. I'll drop the trim when merging.